packages/sanity/src/core/studio/components/navbar/search/SearchDialog.tsx

```diff
@@ -42,148 +39,40 @@ const SearchDialogBox = styled(Box)`
-   * Reset ordering when popover is closed (without valid search terms)
-   */
-  useEffect(() => {
-    if (!hasValidTerms && isMountedRef.current && !open) {
-      dispatch({type: 'ORDERING_RESET'})
-    }
-  }, [dispatch, hasValidTerms, open])
-
-  /**
-   * Set shared `onClose` in search context
-   */
-  useEffect(() => {
-    setOnClose(onClose)
-  }, [onClose, setOnClose])
-
-  /**
-   * Store mounted state
-   */
-  useEffect(() => {
-    if (!isMountedRef?.current) {
-      isMountedRef.current = true
-    }
-  }, [])
-
-  /**
-   * Create a map of indices for our virtual list, ignoring non-filter items.
-   * This is to ensure navigating via keyboard skips over these non-interactive items.
-   */
-  const itemIndices = useMemo(() => {
-    if (hasValidTerms) {
-      return Array.from(Array(result.hits.length).keys())
-    }
-
-    return Array.from(Array(recentSearches.length).keys())
-  }, [hasValidTerms, recentSearches.length, result.hits.length])
-
-  if (!open) {
-    return null
-  }
-
   return (
-    <Portal>
-      <FocusLock autoFocus={false} returnFocus>
-        <CommandListProvider
-          ariaActiveDescendant={itemIndices.length > 0}
-          ariaChildrenLabel={hasValidTerms ? 'Search results' : 'Recent searches'}
-          ariaHeaderLabel="Search"
-          autoFocus
-          data-testid="search-results-dialog"
-          itemIndices={itemIndices}
-          initialSelectedIndex={hasValidTerms ? lastActiveIndex : 0}
-        >
-          <SearchDialogContent
-            filtersVisible={filtersVisible}
-            hasValidTerms={hasValidTerms}
-            onClose={handleClose}
-            open={open}
-          />
-        </CommandListProvider>
-      </FocusLock>
-    </Portal>
-  )
-}
-
-function SearchDialogContent({
-  filtersVisible,
-  hasValidTerms,
-  onClose,
-  open,
-}: {
-  filtersVisible: boolean
-  hasValidTerms: boolean
-  onClose: (lastActiveIndex: number) => void
-  open: boolean
-}) {
-  const {scheme} = useColorScheme()
-  const {getTopIndex} = useCommandList()
-
-  /**
-   * Store top-most search result scroll index on close
-   */
-  const handleClose = useCallback(() => {
-    onClose(getTopIndex())
-  }, [getTopIndex, onClose])
-
-  /**
-   * Bind hotkeys to close action
-   */
-  useSearchHotkeys({onClose: handleClose, open})
-
-  return (
-    <SearchDialogBox>
-      <InnerCard display="flex" height="fill" scheme={scheme} tone="default">
-        <SearchHeader onClose={handleClose} />
-        {filtersVisible && (
-          <Card borderTop style={{flexShrink: 0}}>
-            <Filters />
-          </Card>
-        )}
-
-        <CommandListContainer>
-          {hasValidTerms ? <SearchResults onClose={handleClose} /> : <RecentSearches />}
-        </CommandListContainer>
-      </InnerCard>
-    </SearchDialogBox>
+    <SearchWrapper hasValidTerms={hasValidTerms} onClose={onClose} onOpen={onOpen} open={open}>
+      <Portal>
+        <FocusLock autoFocus={false} returnFocus>
```

Comment:
There is some focus issue with the search text input. When I start typing, the focus is lost when the search results popover appear and the search text input expands. Using `autoFocus` seems to solve the problem, but that might introduce other issues?